pkgs/development/haskell-modules/lib.nix

```diff
@@ -76,6 +76,14 @@ rec {
     fixupPhase = ":";
   });
 
+  # link executables statically against haskell libs to reduce closure size
+  justStaticExecutables = drv: overrideCabal drv (drv: {
+    enableSharedExecutables = false;
+    isLibrary = false;
+    doHaddock = false;
+    postFixup = "rm -rf $out/lib $out/nix-support $out/share/doc";
```

Comment:
Don't manpages end up in `$out/share/man`, I wouldn't think there's anything important in doc for an executable.